src/com/facebook/buck/ide/intellij/IjProjectTemplateDataPreparer.java

```diff
@@ -363,6 +363,16 @@ private boolean isRootAndroidResourceDirectory(IjModule module, Path dir) {
 
     boolean isAndroidFacetPresent = androidFacetOptional.isPresent();
     androidProperties.put("enabled", isAndroidFacetPresent);
+
+    Path basePath = module.getModuleBasePath();
+
+    Optional<Path> extraCompilerOutputPath = projectConfig.getExtraCompilerOutputModulesPath();
+    if (isAndroidFacetPresent
```

Comment:
My mistake. I will remove the 2nd call. In a prior PR (https://github.com/facebook/buck/pull/1412/files#diff-3edfe4c74fe850fc67a4d5eb7422c512R388), I already added the compiler output property to the IML for every Android module. The synthetic modules don't have the Android facet, which is why the check has been updated to accommodate both scenarios